res/controllers/MVave-SMK-25-II-scripts.js

```diff
@@ -0,0 +1,290 @@
+"use strict";
+
+// eslint-disable-next-line no-var
+var SMK25II;
+(function(SMK25II) {
+    class Deck extends components.Deck {
+        constructor() {
+            super([1, 2, 3, 4]);
+            // Knobs
+            this.gainKnob = new components.Pot({
+                group: "[Channel1]",
+                key: "pregain",
+                midi: [0xB0, 0x1E],
+                softTakeover: true,
+            });
+            this.highKnob = new components.Pot({
+                group: "[EqualizerRack1_[Channel1]_Effect1]",
+                midi: [0xB0, 0x1F],
+                key: "parameter3",
+                softTakeover: true,
+            });
+            this.midKnob = new components.Pot({
+                group: "[EqualizerRack1_[Channel1]_Effect1]",
+                midi: [0xB0, 0x20],
+                key: "parameter2",
+                softTakeover: true,
+            });
+            this.lowKnob = new components.Pot({
+                group: "[EqualizerRack1_[Channel1]_Effect1]",
+                midi: [0xB0, 0x21],
+                key: "parameter1",
+                softTakeover: true,
+            });
+            this.effectKnob = new components.Pot({
+                group: "[QuickEffectRack1_[Channel1]]",
+                midi: [0xB0, 0x22],
+                key: "super1",
+                softTakeover: true,
+            });
+            // Hotcues
+            this.hotcues = [];
+            for (let i = 0; i < 8; i++) {
+                const hotcueButton = new components.HotcueButton({
+                    number: i + 1,
+                    group: "[Channel1]",
+                    midi: [0x90, i],
+                    type: components.Button.prototype.types.push,
+                });
+                this.hotcues[i] = hotcueButton;
```

Comment:
Small thing I noticed. why the `type: push`? Also: can you influence the color of the pads by the midi value? It would be great if the color of the pad matched the color of the hotcue.

Reply:
This is the default so it's not strictly needed, just being explicit.
Unfortunately the color can't be set, this is a very cheap controller :)

Reply:
So how does it work then? Because the pads are RGB, right? Does the firmware do all the mode switching? Have you tried to send different velocity values to the pads and see if that changes anything?

Reply:
Yes, it's all handled by the firmware. The color depends on which layer you're on, and you can turn some of them on or off depending on the layer but you can't affect the color

Reply:
Okay thats good to know. Maybe add a comment for that in the manual and the code so people aren't tempted to file issues for it (or waste their time trying to implement it).

Reply:
Its funny to see how cheap controllers just get it wrong all the time. I guarantee it would've been cheaper not to build this functionality into the controller and just let the host handle the lighting, but manufacturers just choose to get it wrong.

Reply:
Done; comment added; thanks!